code/datums/components/rust.dm

```diff
@@ -0,0 +1,83 @@
+/datum/component/rust
+	dupe_mode = COMPONENT_DUPE_HIGHLANDER
+	var/atom/parent_atom
+	var/mutable_appearance/rust_overlay
+	var/static/list/typecache_of_valid_types = list(
+		/turf/closed/wall,
+		/turf/closed/wall/r_wall,
+		/turf/open/floor/plating,
+	)
+
+/turf/closed/wall/rust/New()
+	var/atom/T = new /turf/closed/wall(src)
+	T._AddComponent(list(/datum/component/rust))
```

Comment:
```suggestion
	T.AddComponent(/datum/component/rust)
```

Reply:
I prefer calling the proc over the mscro, is there a reason the macro is preferred?

Reply:
```suggestion
	wall_new._AddComponent(list(/datum/component/rust))
```

Reply:
> I prefer calling the proc over the mscro, is there a reason the macro is preferred?

The proc is internal only.